### PR TITLE
Added backend functionality as well as Modals

### PR DESCRIPTION
**Pull request by Jlane20**

### CODE REVIEW

File: .gitignore
Comment (millerw4) [inline .gitignore:1]: I think we should leave package-lock out of .gitignore. See this gist on package-lock conflict resolution for details [link,](https://gist.github.com/szemate/6fb69c8e3d8cce3efa9a6c922b337d98) but basically, when we get merge conflicts for package-lock we should: 

1. resolve any conflicts in package.json manually
2. ignore conflicts in package-lock.json
3. run npm install
4. if code builds and runs well, continue with merge process as normal

File: package-lock.json
Comment (millerw4) [inline package-lock.json:1]: See above, but please leave this in

File: server/API.js
Comment (millerw4) [inline server/API.js:1]: re: config.token/TOKEN: I've had to juggle between these after pulling as well. Let's try and figure out a team-wide naming scheme for config.token/TOKEN. 

Also, please remove the stray console log on line 6

File: server/index.js
Comment (millerw4) [inline server/index.js:1]: Why is dotenv commented out? Did it break something somewhere?

File: src/Components/QnA/QnA.jsx
Comment (millerw4) [inline src/Components/QnA/QnA.jsx:1]: Please remove the stray console log on line 9

File: src/Components/QnA/QuestionModal.jsx
Comment (millerw4) [inline src/Components/QnA/QuestionModal.jsx:1]: Please remove the stray console log on line 7

**Review by millerw4 — CHANGES_REQUESTED** [review]
Looks good overall! Please remove the couple of stray console logs and see my note about the new strategy to deal with package-lock.json.

File: server/index.js
Comment (Jlane20) [reply server/index.js:1]: I don't have a . env file path so it errored when it tried to require it into the file.

**Pull request closed without merging**